x-pack/plugins/cases/public/containers/user_profiles/use_suggest_user_profiles.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useState } from 'react';
+import { isEmpty } from 'lodash';
+import { useQuery, UseQueryResult } from 'react-query';
+import useDebounce from 'react-use/lib/useDebounce';
+import { UserProfile } from '@kbn/security-plugin/common';
+import { DEFAULT_USER_SIZE } from '../../../common/constants';
+import * as i18n from '../translations';
+import { useKibana, useToasts } from '../../common/lib/kibana';
+import { ServerError } from '../../types';
+import { USER_PROFILES_CACHE_KEY, USER_PROFILES_SUGGEST_CACHE_KEY } from '../constants';
+import { suggestUserProfiles, SuggestUserProfilesArgs } from './api';
+
+const DEBOUNCE_MS = 500;
+
+export const useSuggestUserProfiles = ({
+  name,
+  owner,
+  size = DEFAULT_USER_SIZE,
+}: Omit<SuggestUserProfilesArgs, 'signal' | 'http'>) => {
+  const { http } = useKibana().services;
+  const [debouncedName, setDebouncedName] = useState(name);
+
+  useDebounce(() => setDebouncedName(name), DEBOUNCE_MS, [name]);
+
+  const toasts = useToasts();
+
+  return useQuery<UserProfile[], ServerError>(
+    [
+      USER_PROFILES_CACHE_KEY,
+      USER_PROFILES_SUGGEST_CACHE_KEY,
+      { name: debouncedName, owner, size },
+    ],
+    () => {
+      const abortCtrlRef = new AbortController();
+      return suggestUserProfiles({
+        http,
+        name: debouncedName,
+        owner,
+        size,
+        signal: abortCtrlRef.signal,
+      });
+    },
+    {
+      enabled: !isEmpty(debouncedName),
```

Comment:
hmm does this mean we won't show anything if the user hasn't typed anything? I wonder what Elasticsearch will return if we don't send in a `name`, maybe the most recently logged in users? I can test that.

Reply:
Yes, that means that the query will not run if the user hasn't typed anything. You are right. Let's test it together offline and see what the results will be.

Reply:
Ok, I tested it and ES returns a suggestion of users even with an empty string. I remove the `enabled` to simulate the same behavior.